src/main/java/org/folio/services/ftp/SftpUploadService.java

```diff
@@ -58,20 +59,29 @@ private ApacheSshdSftpSessionFactory getSshdSessionFactory(String username, Stri
   public Future<String> upload(Context ctx, String username, String password, String folder, String filename, String content)
       throws Exception {
     Promise<String> promise = Promise.promise();
-    String folderPath = StringUtils.isEmpty(folder) ? "" : (folder + FILE_SEPARATOR);
-    String remoteAbsPath = folderPath + filename;
-
+    String remoteAbsPath;
+    if (StringUtils.isNotEmpty(folder)) {
+      String folderPath = StringUtils.isEmpty(folder) ? "" : (folder + FILE_SEPARATOR);
+      remoteAbsPath = folderPath + filename;
+    } else {
+      remoteAbsPath = DEFAULT_WORKING_DIR + FILE_SEPARATOR + filename;
+    }
     SessionFactory<SftpClient.DirEntry> sshdFactory;
+    Session<SftpClient.DirEntry> session;
     try {
       sshdFactory = getSshdSessionFactory(username, password);
+      session = sshdFactory.getSession();
```

Comment:
In this case session is not closed by anyone, correct?

Reply:
I see now, you created it before try with resources block and use in try with resources that should close it